bindings/java/java/src/main/java/org/ethereum/evmc/EvmcVm.java

```diff
@@ -7,37 +7,24 @@
 import static org.ethereum.evmc.Host.removeContext;
 
 import java.nio.ByteBuffer;
-import java.util.Objects;
 
 /**
  * The Java interface to the evm instance.
  *
  * <p>Defines the Java methods capable of accessing the evm implementation.
  */
 public final class EvmcVm implements AutoCloseable {
-  private static EvmcVm evmcVm;
-  private static boolean isEvmcLibraryLoaded = false;
   private ByteBuffer nativeVm;
+
   /**
    * This method loads the specified evm shared library and loads/initializes the jni bindings.
    *
+   * @param evmcPath the path to the evmc shared library
    * @param filename /path/filename of the evm shared object
    */
-  public static EvmcVm create(String filename) {
-    if (!EvmcVm.isEvmcLibraryLoaded) {
-      try {
-        // load so containing the jni bindings to evmc
-        System.load(System.getProperty("user.dir") + "/../c/build/lib/libevmc-java.so");
-        EvmcVm.isEvmcLibraryLoaded = true;
-      } catch (UnsatisfiedLinkError e) {
-        System.err.println("Native code library failed to load.\n" + e);
-        System.exit(1);
-      }
-    }
-    if (Objects.isNull(evmcVm)) {
-      evmcVm = new EvmcVm(filename);
-    }
-    return evmcVm;
+  public static EvmcVm create(String evmcPath, String filename) {
+    System.load(evmcPath);
+    return new EvmcVm(filename);
```

Comment:
allow to reuse a VM, so remove the static singleton, remove System.exit, and just load the evmc with a file path to the evmc native library.